server/src/handler/random.rs

```diff
@@ -0,0 +1,50 @@
+use tpm2_rs_base::errors::TpmRcError;
+
+use crate::{
+    handler::CommandHandler,
+    platform::{
+        crypto::{Drbg, EntropySource},
+        TpmBuffers, TpmContextDeps,
+    },
+    req_resp::RequestThenResponse,
+    ServerError,
+};
+
+impl<Deps: TpmContextDeps> CommandHandler<Deps> {
+    fn try_get_random(&mut self, buffer: &mut [u8]) -> Result<(), ServerError<Deps>> {
+        if self.crypto.drbg.requires_reseeding() {
+            let mut seed = <Deps::Drbg as Drbg>::Entropy::default();
+            self.crypto.entropy.fill_entropy(seed.as_mut());
+            self.crypto
+                .drbg
+                .reseed(&seed, &[])
+                .map_err(ServerError::<Deps>::Drbg)?;
```

Comment:
You should just implement `From<DrbgError> for ServerError`, so that this mapping happens for you via the try-operator.